wagtail/images/fields.py

```diff
@@ -42,27 +42,25 @@ def __init__(self, *args, **kwargs):
             }
 
         # Error messages
-        self.error_messages["invalid_image_extension"] = (
-            _("Not a supported image format. Supported formats: %s.")
-            % SUPPORTED_FORMATS_TEXT
-        )
+        self.error_messages["invalid_image_extension"] = _(
+            "Not a supported image format. Supported formats: %(supported_formats)s."
+        ) % {"supported_formats": SUPPORTED_FORMATS_TEXT}
 
         self.error_messages["invalid_image_known_format"] = _(
-            "Not a valid .%s image. The extension does not match the file format (%s)"
+            "Not a valid .%(extension)s image. The extension does not match the file format (%(image_format)s)"
         )
 
-        self.error_messages["file_too_large"] = (
-            _("This file is too big (%%s). Maximum filesize %s.") % max_upload_size_text
-        )
+        self.error_messages["file_too_large"] = _(
+            "This file is too big (%%(file_size)s). Maximum filesize %(max_filesize)s."
```

Comment:
Slightly expanding the scope of the task here, but it would be great if we could eliminate the hacky two-pass string formatting that's going on here (and the file_too_many_pixels message immediately below - I think these two are the only occurrences of this in the codebase).

Since Transifex can't recognise the `%%`-escaped variable as a placeholder (or at least, it couldn't around the time of 4.0 - I didn't see the problem recur in 4.1), we've had multiple cases of user error where translators have done things like inserting whitespace between the `%%` and `s`. (And, of course, adding the variable name here will be one more thing they can potentially get wrong...)

I think this can be fixed by leaving out the `% {"max_filesize": max_upload_size_text}` step here, and inserting that at the same time as `file_size`.